### PR TITLE
Scoring/pricing hdf5

### PR DESCRIPTION
**Pull request by ke4920**
# Description

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: VOLTA/data/entsoe/crawler.py [hunk 1]
```diff
@@ -305,6 +305,53 @@ def load_hdf5(file_path: Union[pathlib.Path, str]) -> None:
...
+    file_path : pathlib.Path | str
+        The full path to the HDF5 file, including file name.
+    group : str
+        The name of the dataset group, e.g., "nodes" or "edges".
+    index : int
+        The index of the dataset in its group.
+
+    Returns
+    -------
+    pd.DataFrame | pd.Series
```
Comment (mcw92) [inline VOLTA/data/entsoe/crawler.py:325]:
```suggestion
    pd.DataFrame
```
See above.

File: VOLTA/data/entsoe/hdfwriter.py [hunk 2]
```diff
@@ -84,6 +85,48 @@ def get_bidding_zones_with_data(self) -> None:
...
+            index = pd.date_range(
+                year, self.intervals[idx + 1], freq=self.frequency, tz="UTC"
+            )
+            if path_to_pricing_file.is_file():
+                pricing = pd.read_csv(path_to_pricing_file, index_col=0)
+                pricing.index = pd.to_datetime(pricing.index, utc=True)
+                pricing = pricing.reindex(index)
+            else:
+                pricing = pd.DataFrame(np.nan, index=index, columns=["price"])
+                log.debug("No file present, interpret data as Nan")
```
Comment (mcw92) [inline VOLTA/data/entsoe/hdfwriter.py:122]:
```suggestion
                log.debug("No file present, interpret data as NaN.")
```
Sorry for this one 🙈. I kind of have an eye for random, not so important details like this. However, consistency is always nice 😌.

File: VOLTA/data/entsoe/hdfwriter.py [hunk 3]
```diff
@@ -185,6 +228,55 @@ def get_edge_features_from_csv(
...
+                    f"Pricing for {node} contain {pricing_data.notna().sum().sum()} non-NaN values."
+                )
+                starttime = pricing_data.index[0].strftime("%Y-%m-%d %H:%M:%S%z")
+                endtime = pricing_data.index[-1].strftime("%Y-%m-%d %H:%M:%S%z")
+                if pricing_data.shape[0] != 82536:
+                    log.debug(f"AAAAAAAAAAAAAAAA: Node {node} is fucked up =(!")
+
+                log.debug(f"Pricing data shape is {pricing_data.shape}.")
+                log.debug(
+                    f"Number of non-NaNs in pricing data {node} is {pricing_data.notna().sum().sum()}"
```
Comment (mcw92) [inline VOLTA/data/entsoe/hdfwriter.py:267]:
```suggestion
                    f"Number of non-NaNs in pricing data {node} is {pricing_data.notna().sum().sum()}."
```

File: tests/data/entsoe_pricing/test_pricing_hdfwriter.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+
+def test_entsoe_hdfwriter_one_bdding_zone() -> None:
+    """Test the conversion from csv data into hdf5 format for one bidding-zone."""
+    test_root = pathlib.Path(__file__).resolve().parent
+    csv_path = test_root / "test_data"
+
+    output_file = test_root / "entsoe_pricing_test.h5"
+
+    # Test data was downloaded for the following years and bidding-zones:
```
Comment (mcw92) [inline tests/data/entsoe_pricing/test_pricing_hdfwriter.py:60]:
```suggestion
    # Test data was downloaded for the following years and bidding zones:
```

**Review by mcw92 — APPROVED** [review]
I went through all the code and it looks code. I made a couple of comments including some minor commit suggestions which should be included before merging. Otherwise, this can be merged in my view 👍.

File: VOLTA/data/entsoe/hdfwriter.py [hunk 5]
```diff
@@ -185,6 +228,55 @@ def get_edge_features_from_csv(
...
+                log.debug(
+                    f"Number of non-Nans in array is {pricing_data.to_numpy().size - np.isnan(pricing_data.to_numpy()).sum()}."
+                )
+                file[f"nodes/{idx}"] = pricing_data.to_numpy(dtype=np.float32)
+
+            file["metadata"].attrs["starttime"] = starttime
+            file["metadata"].attrs["endtime"] = endtime
+            file["metadata"].attrs["frequency"] = self.frequency
+            file["metadata"].attrs["node_feature_list"] = ["price"]
+            file["metadata"].attrs["node_feature_units_list"] = ["euro"]
```
Comment (JuanPedroGHM) [inline VOLTA/data/entsoe/hdfwriter.py:278]: Feature list to "day_ahead_prices" and the unit to "euro/MWh" to keep it a bit more correct

**Review by JuanPedroGHM — CHANGES_REQUESTED** [review]
Small change needed in the hdf5 metadata. Other than that it is great

File: VOLTA/data/entsoe/hdfwriter.py (hunk 5 above)
Comment (ke4920) [reply VOLTA/data/entsoe/hdfwriter.py]: ok, good point, i updated the meta-data, and the test

**Review by JuanPedroGHM — APPROVED** [review]

**Pull request merged**